testsupport/wait/awaitility.go

```diff
@@ -185,10 +185,17 @@ func (a *Awaitility) WaitForNamedToolchainClusterWithCondition(name string, cond
 // if the CR has the ClusterCondition
 func (a *Awaitility) GetToolchainCluster(clusterType cluster.Type, namespace string, condition *toolchainv1alpha1.ToolchainClusterCondition) (toolchainv1alpha1.ToolchainCluster, bool, error) {
 	clusters := &toolchainv1alpha1.ToolchainClusterList{}
-	if err := a.Client.List(context.TODO(), clusters, client.InNamespace(a.Namespace), client.MatchingLabels{
+	matchingLabels := client.MatchingLabels{
 		"namespace": namespace,
 		"type":      string(clusterType),
-	}); err != nil {
+	}
+	// add cluster role label tenant as filter for member type clusters
+	// this covers the scenario in which the `tenant` role label for member cluster should be set.
+	if clusterType == cluster.Member {
+		// we use only the label key, the value can remain empty.
+		matchingLabels[cluster.RoleLabel(cluster.Tenant)] = ""
+	}
+	if err := a.Client.List(context.TODO(), clusters, client.InNamespace(a.Namespace), matchingLabels); err != nil {
```

Comment:
we could actually verify that the `tenant` label is not set for the host and e2e ToolchainClusters in this part.
I mean - at the end of this `GetToolchainCluster` method :-)

Reply:
given that `checkTenantRoleLabel` will already check that the label is adding correctly does it still make sense to have the check also here ? 
Maybe I can remove the label filtering at this point ... WDYT ?

Reply:
yup, makes sense  :+1: